### DISCUSSION TITLE
Sponge download

### DISCUSSION THREAD

**Discussion by mosemister** [ArcanePlugins/Treasury#310] [Q&A] [answered]
Im just curious about 3 things about the Sponge edition

- Why isn't it on Sponge Ore? 
- How does Treasury compare to the built in economy service? 
- (if its not obvious from the previous question) why would a developer making a sponge exclusive plugin not target the sponge built in economy but instead treasury?

**Comment by thrailkillr** [comment]
> Im just curious about 3 things about the Sponge edition
> 
> * Why isn't it on Sponge Ore?
> * How does Treasury compare to the built in economy service?
> * (if its not obvious from the previous question) why would a developer making a sponge exclusive plugin not target the sponge built in economy but instead treasury?

1. I have no clue why they haven't uploaded it yet.
2 & 3. Treasury has a general-purpose **asynchronous** economy api, allowing transactions to be able to cancelled by the provider.

**Reply by mosemister** [reply]
Thanks for the 1)

As for "allowing transactions to be able to be called by the provider". Wouldnt listening to sponge's [Transaction Event](https://jd.spongepowered.org/spongeapi/10.0.0/org/spongepowered/api/event/economy/EconomyTransactionEvent.html) achieve the same?

But i get the completely asynced helps offload from main thread allowing for sql friendly transactions to happen (when i wrote this originally i had it in my mind that sponge's economy was async compatible) 

Is there any others then that?

**Comment by MrIvanPlays** [comment, marked as answer]
Its not on Sponge Ore because no one quite literally mentioned to have an upload there. Benefits of treasury were that everything is done asynchronously and you can utilize it to make your plugin cross-platform and save up code. Since no adoption happened due to lack of technical knowledge of java amongst most of the plugin developers (and vault still being the go-to API for developing economy plugins) the project stays as-is... dead...
